typescript/sdk/src/deploy/HyperlaneAppChecker.ts

```diff
@@ -106,15 +97,36 @@ export abstract class HyperlaneAppChecker<
             provider,
             contract.address,
           );
-          if (!eqAddress(actualProxyAdminAddress, expectedProxyAdminAddress)) {
+          const actualProxyAdminContract = ProxyAdmin__factory.connect(
+            actualProxyAdminAddress,
+            provider,
+          );
+          const actualProxyAdminOwner = await actualProxyAdminContract.owner();
+          const expectedOwner = ownableOverrides?.['proxyAdmin'] ?? owner;
+          if (!eqAddress(actualProxyAdminOwner, expectedOwner)) {
```

Comment:
Here we check if the `proxyAdmin` owner is the expected config owner. This will do this check for AW owned and non AW owned `proxyAdmins`. 

We need to do this here because for non AW owned `proxyAdmins`, we don't have the `proxyAdmins` address in config and the explicit ownership check will therefore not be done in `checkOwnership`

This does me that the ownership check for AW owned `proxyAdmin` is done twice but I think this is unavoidable and not a high cost

Reply:
Even with these comments it took me a bit to understand what was going on here, could you add comments to the code for future readers? e.g. describing why `expectedProxyAdminAddress` may be undefined, why we have all this logic, etc